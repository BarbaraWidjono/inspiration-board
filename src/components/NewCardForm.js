import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props){
    super(props);

    this.state = {
      text: "",
      emoji: "",
    };
  }

  inputChange = (event) => {
    const field = event.target.name;
    const value = event.target.value;

    const newState = {}
    newState[field] = value;
    this.setState(newState);
  }

  onFormSubmit = (event) => {
    event.preventDefault();

    const newNote = {
      text: this.state.text,
      emoji: this.state.emoji,
    }

    if(this.state.text.length > 0){
      this.setState({
        text: "",
        emoji: "",
      })
    }

    this.props.createNoteCallback(newNote);
  }

  render(){
    return(
      <form
        className="new-card-form"
        onSubmit={this.onFormSubmit}>
        <h3 className="new-card-form__header">Add a Note</h3>
        <div>
          <label htmlFor="text" className="new-card-form__form-label">Text:</label>
          <input
            name="text"
            value={this.state.text}
            onChange={this.inputChange}
          />
        </div>
        <div>
          <label htmlFor="emoji" className="new-card-form__form-label">Emoji:</label>
          <input
            name="emoji"
            value={this.state.emoji}
            onChange={this.inputChange}
          />
        </div>
        <input type="submit" name="submit" value="Create New Note" />
      </form>
    )
  }
}

export default NewCardForm;
