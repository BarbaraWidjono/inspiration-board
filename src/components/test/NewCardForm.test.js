import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('matches existing snapshot', () => {
    const wrapper = shallow( <NewCardForm createNoteCallback={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  })
})
