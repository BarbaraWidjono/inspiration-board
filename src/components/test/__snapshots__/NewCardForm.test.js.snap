// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm matches existing snapshot 1`] = `
<form
  className="new-card-form"
  onSubmit={[Function]}
>
  <p
    className="new-card-form__header"
  >
    Add a Note
  </p>
  <section
    className="new-card-form__form"
  >
    <div>
      <label
        className="new-card-form__form-label"
        htmlFor="text"
      >
        Text:
      </label>
      <input
        name="text"
        onChange={[Function]}
        value=""
      />
    </div>
    <div>
      <label
        className="new-card-form__form-label"
        htmlFor="emoji"
      >
        Emoji:
      </label>
      <input
        name="emoji"
        onChange={[Function]}
        value=""
      />
    </div>
    <input
      className="new-card-form__form-button"
      name="submit"
      type="submit"
      value="Create New Note"
    />
  </section>
</form>
`;
